client_code/atomic/atoms.py

```diff
@@ -114,6 +114,16 @@ def portable_atom(_cls, name=None):
         _cls.__serialize__ = lambda self, _: {
             k: v for k, v in self.__dict__.items() if k != REGISTRAR
         }
+
+    if not any(
+        [hasattr(_cls, attr) for attr in ("__deserialize__", "__new_deserialized__")]
```

Comment:
no need for the `[`?

Reply:
any needs some form of iterable, so I went with a list comprehension.

Do you have some other preference?

Reply:
Without the square brackets it's just a generator no? Saves us doing two loops. One for the list comp and one for the call to any.

Reply:
sorry if I'm being a bit thick here but I don't quite get what you're saying.

`any` is a function that takes a single iterable argument.
I'm using a list comprehension as that iterable

What am I missing here?

Reply:
```python
>>> any(i for i in [1,2,3])
True
>>> any(i for i in [0, 0, 0])
False
```

```python
if not any(hasattr(_cls, attr) for attr in ("__deserialize__", "__new_deserialized__")):
   ...
```

Reply:
ahhhh!!! I see. Told you I was being a bit thick!